.changeset/khaki-glasses-raise.md

```diff
@@ -0,0 +1,48 @@
+---
+'astro': minor
+---
+
+## Integration Hooks to add Middleware
+
+It's now possible in Astro for an integration to add middleware on behalf of the user. Previously when a 3rd party wanted to provide middleware the user would need to create a `src/middleware.ts` file themselves. Now adding 3rd party middleware is as easy as adding a new integration.
+
+For integration authors, there is a new `addMiddleware` function in the `astro:config:setup` hook. This function allows you to specify a middleware module and the order in which it should be applied:
+
+__my-package/middleware.js__
+
+```js
+import { defineMiddleware } from 'astro:middleware';
+
+export const onRequest = defineMiddleware(async (context, next) => {
+  const response = await next();
+
+  if(response.headers.get('content-type') === 'text/html') {
+    let html = await response.text();
+    html = minify(html);
+    return new Response(html, {
+      status: response.status,
+      headers: response.headers
+    });
+  }
+
+  return response;
+});
+```
+
```

Comment:
```suggestion

You can now add your integration's middleware either before or after...
```
Not super necessary here, but more as a note for the upcoming blog post that it's always better to have text in between code samples so that you can call attention to what the important thing is about each one.

Also, I saw the discussion on `order` and that the options seem to be `pre` or `post`. If it's something that sparked discussion in the PR, that usually means it's good fodder for the blog post to call out, add context, or just to make moar coNTenT!

So, a line here about "before or after" in at least the blog post I think makes a lot of sense!